### DISCUSSION TITLE
Stdout redirection

### DISCUSSION THREAD

**Discussion by keastrid** [jnr/jnr-ffi#324] [General]
Is there any way to get C print statements, eg. printf, to show up in java stdout?

It would make debugging much easier to see what is being passed to and going on in native methods.

**Comment by basshelal** [comment]
Actually, the Java process should be using the same `stdout` as the native code, your issue may be related to buffer flushing of `stdout` before writing on the native side, which will often lead to native code's `printf` statements showing up long after they are expected because they were in a buffer and weren't written to the output in time.

A solution is to use an unbuffered output like `stderr` which would show up instantly, otherwise, you may have to manually call `fflush` either from native code or in Java (using a mapping) in order to show the latest updates to the output buffer. I believe C also has a way to unbuffer `stdout` though that may also affect the Java's `stdout` too (because it's the `stdout` of the process) so that may not be ideal.
